dashboard/design/survey_summary_diagram.puml

```diff
@@ -0,0 +1,143 @@
+@startuml
+
+left header
+  <size:14>Last updated: 4/18/19
+  <size:14>Work in progress
+end header
+
+title
+  <color:#blue><size:18>**JotForm Survey Summary Assembly Line**
+end title
+
+start
+
+partition 1.Ingest {
+  #lightgrey:Jotform submissions
+  (in Jotform server)]
+  :Fetcher;
+  note right
+    Sync data from Jotform to CDO db as-is.
+  end note
+  #lightgrey:submissions_table
+  (in CDO db)]
+  note right #lightgrey
+    Schema: (submission_json_str)
+    Only access sequentially.
+    Only insert to this table, no update.
+  end note
+}
+
+partition 2.Query {
+  :Parser;
+  note right
+    Parse submission JSON string, expose key information to enable direct access
+    to submission content using submission id.
+  end note
+  #lightgrey:submissions_parsed_table]
+  note right #lightgrey
+    Schema: (submission_id, form_id, created_at, status, submission_json_str)
+    Only insert to this table, no update.
+  end note
+
+  :Indexer;
+  note right: Index submission using common lookup keys such as workshop_id
+  #lightgrey:submissions_lookup_table]
+  note right #lightgrey
+    Schema: (submission_id, form_id, environment, workshop_id, user_id, facilitator_id)
+    Can be destroyed and rebuilt if schema changes.
+  end note
+
+  :Filter;
+  note right
+    Filter submissions based on lookup conditions (e.g. CSF 201 in 2019)
+    and validation rules (e.g. ignore invalid submissions)
+  end note
+  #lightgrey:submission_id_array]
+}
+
+partition 3.Summarize {
+  :Retriever;
+  note right: Retrieve submission content from database to in-memory data structure.
+  #lightgrey:submission_content_array]
+  note right #lightgrey
+    [{submission_id, form_id, answers}] (array of hash)
+    answers = {question_id : question_and_answer}
+  end note
+
+  :Modifier;
+  note right
+    Modify question and answer data to make them aggregatable.
+    E.g. update question unique name or convert value range to make values consistent.
```

Comment:
I'd like more detail on this step.

- How does the system know what modifications to perform?
- Who writes these modification rules?
- Are we encoding these rules as checked-in code, or is this configuration in the same way that survey ids are configuration?
- When are the rules written, or how do we find out that new rules need to be written? Is there any user impact in a possible delay here?
- How do these rules propagate across environments?
- Can we present some specific example rules?

Reply:
Added to Appendix.